pkg/grpc/marshaler.go

```diff
@@ -0,0 +1,67 @@
+package grpc
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/protoadapt"
+)
+
+// Deprecated: customMarshaler is a hack for keeping backward compatibility for duration objects.
+// TODO(ROX-25678): This should be deleted in 4.8.
+type customMarshaler struct {
+	*runtime.JSONPb
+}
+
+func (c customMarshaler) Unmarshal(data []byte, v interface{}) error {
```

Comment:
Consider pointer receiver:
```suggestion
func (c *customMarshaler) Unmarshal(data []byte, v interface{}) error {
```